### PR TITLE
AP_Logger: address review comments from #14299

### PR DESCRIPTION
**Pull request by andyp1per**

### CODE REVIEW

File: libraries/AP_Logger/AP_Logger_Backend.cpp [hunk 1]
```diff
@@ -530,7 +530,7 @@ bool AP_Logger_Backend::logging_enabled() const
 
 void AP_Logger_Backend::Write_AP_Logger_Stats_File(const struct df_stats &_stats)
 {
-    struct log_DSF pkt = {
+    const struct log_DSF pkt = {
```
Comment (peterbarker) [inline libraries/AP_Logger/AP_Logger_Backend.cpp:533]: Lose the `=` sign here.

File: Tools/autotest/common.py [hunk 2]
```diff
@@ -5386,7 +5386,7 @@ def validate_log_file(self, logname, header_errors=0):
         class Capturing(list):
             def __enter__(self):
                 self._stderr = sys.stderr
-                sys.stderr = self._stringio = StringIO()
+                sys.stderr = self._stringio = StringIO.StringIO()
```
Comment (peterbarker) [inline Tools/autotest/common.py:5389]: Works on both Python2 and Python3 now?

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_Logger/AP_Logger_Backend.cpp (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_Logger/AP_Logger_Backend.cpp:533]: Fixed

File: Tools/autotest/common.py (hunk 2 above)
Comment (andyp1per) [reply Tools/autotest/common.py:5389]: Yes

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**